### PR TITLE
Add Ollama provider

### PR DESCRIPTION
**Pull request by antaz**
## Changes
- Add Ollama provider

### CODE REVIEW

**Comment by 0x1eef** [comment]
This is cool to see and I'm really excited about adding this provider. I tried locally and ran into this error:
```
=> "{\"error\":\"json: cannot unmarshal string into Go struct field ChatRequest.stream of type bool\"}"
```
Given:
```
llm = LLM.ollama("")
bot = llm.chat("hello")
```

File: lib/llm/providers/ollama.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+module LLM
+  ##
+  # The Ollama class implements a provider for
+  # [Ollama](https://ollama.ai/)
+  class Ollama < Provider
+    require_relative "ollama/error_handler"
+    require_relative "ollama/response_parser"
+
+    HOST = "localhost"
+    DEFAULT_PARAMS = {model: "llama3.2", stream: "false"}.freeze
```
Comment (0x1eef) [inline lib/llm/providers/ollama.rb:12]: I think "false" should be `false` (bare, no quotes)

File: lib/llm.rb [hunk 2]
```diff
@@ -28,6 +28,14 @@ def gemini(secret)
     LLM::Gemini.new(secret)
   end
 
+  ##
+  # @param host (see LLM::Ollama#initialize)
+  # @return (see LLM::Ollama#initialize)
+  def ollama(secret)
```
Comment (0x1eef) [inline lib/llm.rb:34]: What is `secret` suppose to be ? Seems to work without one ?

Comment (antaz) [reply lib/llm.rb:34]: @0x1eef Ollama models are deployed without a secret i think, Can this a convincing case to make secret optional too? :)

File: lib/llm/provider.rb [hunk 3]
```diff
@@ -14,10 +14,10 @@ class Provider
     #  The host address of the LLM provider
     # @param [Integer] port
     #  The port number
-    def initialize(secret, host, port = 443)
+    def initialize(secret, host, port = 443, ssl: true)
```
Comment (0x1eef) [inline lib/llm/provider.rb:17]: Can we make `host`, and `port` keyword arguments too ? Everything given to us via `options` should be a keyword argument.

File: lib/llm/providers/ollama.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    require_relative "ollama/error_handler"
+    require_relative "ollama/response_parser"
+
+    HOST = "localhost"
+    DEFAULT_PARAMS = {model: "llama3.2", stream: false}.freeze
+
+    ##
+    # @param secret (see LLM::Provider#initialize)
+    def initialize(secret, **)
+      super(secret, HOST, 11434, ssl: false, **)
```
Comment (0x1eef) [inline lib/llm/providers/ollama.rb:17]: This looks weird to me. Why are we using 3 positional arguments, and one keyword argument ? The interface is this:
```
LLM.open(secret, options)
```
Everything that goes into `options` should be a keyword argument.

**Review by 0x1eef — COMMENTED** [review]
The method signature for the providers doesn't look right to me. We shouldn't be using positonal arguments for options. Those should be keyword arguments. Here's the interface:
```
LLM.openai(secret, options)
```
And options is a key/value pairs, so it doesn't make sense to turn them back into positional arguments.

File: lib/llm/providers/ollama.rb (hunk 4 above)
Comment (0x1eef) [inline lib/llm/providers/ollama.rb:17]: I don't think this does what you expect. Keyword arguments won't "override" positional arguments. So, the syntax here `**` will have no effect on the host and port - right ? Let's not use syntax we don't understand, and revert back to something more simple:

```
    def initialize(secret, **options)
      super(secret, **{host: HOST}.merge(options))
   end
```

File: lib/llm.rb [hunk 5]
```diff
@@ -15,24 +15,32 @@ module LLM
...
   # @return (see LLM::Gemini#initialize)
-  def gemini(secret)
+  def gemini(secret, **)
     require_relative "llm/providers/gemini" unless defined?(LLM::Gemini)
-    LLM::Gemini.new(secret)
+    LLM::Gemini.new(secret, **)
+  end
+
+  ##
+  # @param host (see LLM::Ollama#initialize)
```
Comment (0x1eef) [inline lib/llm.rb:32]: Should this not be ...

```
@param host (see LLM::Provider#initialize)
@param port (see LLM::Provider#initialize)
@param ssl (see LLM::Provider#initialize)
```

File: lib/llm.rb (hunk 2 above)
Comment (0x1eef) [reply lib/llm.rb:34]: Thanks!

**Review by 0x1eef — APPROVED** [review]
Great work.

**Pull request merged**